digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/CalibrationPointsUtils.scala

```diff
@@ -118,4 +122,17 @@ object CalibrationPointsUtils {
     }
   }
 
+  def createCalibrationPointIfNeeded(rwPoint: Long, linkId: Long, calibrationPointLocation: CalibrationPointLocation,
+                                     calibrationPointType: CalibrationPointType, username: String): Unit = {
+    val existing = CalibrationPointDAO.fetch(linkId, calibrationPointLocation.value)
+    val (existingCorrect, existingDifferent) = existing.partition(cp => cp.roadwayPointId == rwPoint && cp.typeCode == calibrationPointType)
+    if (existingDifferent.nonEmpty) {
+      CalibrationPointDAO.expireById(existing.map(_.id))
```

Comment:
Found a bug here after merging: I should have expired "existingDifferent" and not all the "existing".